### PR TITLE
Remove -fpic from Windows CUDA issue with the build for ipc-toolkit with vcpkg on Windows 

### PR DESCRIPTION
**Pull request by antoinebou12**
-- Building for: Visual Studio 17 2022
-- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.22631.
-- The CXX compiler identification is MSVC 19.41.34120.0
-- Looking for a CUDA compiler
-- Looking for a CUDA compiler - C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v12.6/bin/nvcc.exe
-- The CUDA compiler identification is NVIDIA 12.6.68

image

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -165,11 +165,13 @@ if(SCALABLE_CCD_WITH_CUDA)
     message(FATAL_ERROR "No CUDA support found!")
   endif()
 
+  # Enable separable compilation for CUDA
+  set_target_properties(scalable_ccd PROPERTIES CUDA_SEPARABLE_COMPILATION ON)
```
Comment (zfergus) [inline CMakeLists.txt:169]: Revert this change as it only moved the line.

File: CMakeLists.txt [hunk 2]
```diff
@@ -165,11 +165,13 @@ if(SCALABLE_CCD_WITH_CUDA)
...
+
   # We need to explicitly state that we need all CUDA files in the particle
+  # Enable separable compilation for CUDA
```
Comment (zfergus) [inline CMakeLists.txt:172]: Delete this duplicate comment.

File: CMakeLists.txt [hunk 3]
```diff
@@ -189,21 +191,34 @@ if(SCALABLE_CCD_WITH_CUDA)
 
   if(APPLE)
     # We need to add the path to the driver (libcuda.dylib) as an rpath,
+    # Add rpath for CUDA on macOS
```
Comment (zfergus) [inline CMakeLists.txt:194]: Delete duplicate comment

**Review by zfergus — CHANGES_REQUESTED** [review]

**Comment by Q-Minh** [comment]
The problem is not with the platform, it's with how the flags are passed to the underlying compiler (nvcc). Position independent code is not specific to Windows. Just set the [`POSITION_INDEPENDENT_CODE`](https://cmake.org/cmake/help/latest/prop_tgt/POSITION_INDEPENDENT_CODE.html) property to ON for the relevant target, or set the global [`CMAKE_POSITION_INDEPENDENT_CODE`](https://cmake.org/cmake/help/latest/variable/CMAKE_POSITION_INDEPENDENT_CODE.html#variable:CMAKE_POSITION_INDEPENDENT_CODE). CMake will know not to add the `-fPIC` flag when nvcc is invoked versus when the native build tool is invoked.

File: CMakeLists.txt [hunk 4]
```diff
@@ -166,10 +166,12 @@ if(SCALABLE_CCD_WITH_CUDA)
   endif()
 
   # We need to explicitly state that we need all CUDA files in the particle
+  # Enable separable compilation for CUDA
   # library to be built with -dc as the member functions could be called by
   # other libraries and executables.
-  set_target_properties(scalable_ccd PROPERTIES CUDA_SEPARABLE_COMPILATION ON)
-
+  set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```
Comment (zfergus) [inline CMakeLists.txt:172]: This is already set to ON in this file.

File: CMakeLists.txt [hunk 5]
```diff
@@ -166,10 +166,12 @@ if(SCALABLE_CCD_WITH_CUDA)
...
+  set_target_properties(scalable_ccd PROPERTIES CUDA_SEPARABLE_COMPILATION ON POSITION_INDEPENDENT_CODE ON)
```
Comment (zfergus) [inline CMakeLists.txt:173]: This does not need to be set as the global option is already set to ON

**Review by zfergus — CHANGES_REQUESTED** [review]

**Review by zfergus — APPROVED** [review]

**Pull request merged**